### PR TITLE
docs(loading): show how to use dismiss method returns from useIonLoading for react

### PR DESCRIPTION
**Pull request by omar2205**
The docs don't say how to use dismiss(), this shows that the `useIonLoading` hook returns the present and dismiss functions.

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
The docs mention quickly using dismiss() to close the IonLoading, but it doesn't say how to access this function.

Issue Number: N/A

## What is the new behavior?
- Just adds the dismiss as a second return in the react example.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Thanks for the PR! To make this change, you should add the `dismiss` line to `core/src/components/loading/usage/react.md` and then run `npm run build` to have that get added to `readme.md`. The part of the `readme.md` file you edited is automatically generated and should not be edited manually.

File: core/src/components/loading/usage/react.md [hunk 1]
```diff
@@ -7,17 +7,18 @@ import { IonButton, IonContent, IonPage, useIonLoading } from '@ionic/react';
...
+  const [present, dismiss] = useIonLoading();
   return (
     <IonPage>
       <IonContent>
         <IonButton
           expand="block"
-          onClick={() =>
+          onClick={() => {
             present({
-              duration: 3000,
```
Comment (liamdebeasi) [inline core/src/components/loading/usage/react.md:18]: I would rather we keep the original `duration: 3000` property. I don't want users thinking the recommended way to dismiss `IonLoading` is to use a `setTimeout` manually when there's a property built in that will do that.

File: core/src/components/loading/usage/react.md [hunk 2]
```diff
@@ -7,17 +7,18 @@ import { IonButton, IonContent, IonPage, useIonLoading } from '@ionic/react';
 interface LoadingProps {}
 
 const LoadingExample: React.FC<LoadingProps> = () => {
-  const [present] = useIonLoading();
+  const [present, dismiss] = useIonLoading();
```
Comment (liamdebeasi) [inline core/src/components/loading/usage/react.md:10]:
```suggestion
  /**
   * The recommended way of dismissing is to use the `dismiss` property
   * on `IonLoading`, but the `dismiss` method returned from `useIonLoading` 
   * can be used for more complex scenarios.
   */
  const [present, dismiss] = useIonLoading();
```

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Looks great. I added a few comments below which should help make the docs a bit clearer.

**Comment by omar2205** [comment]
>   /**
>    * The recommended way of dismissing is to use the `dismiss` property
>    * on `IonLoading`, but the `dismiss` method returned from `useIonLoading` 
>    * can be used for more complex scenarios.
>    */

Did you mean "duration" property?

File: core/src/components/loading/usage/react.md (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components/loading/usage/react.md:18]: I think GitHub may have accidentally marked this as resolve, but I would still like to see this issue resolved.

One idea is that maybe we could add an additional example underneath the one that uses the `duration` property to show how you can manually dismiss the loading component? You could use your `setTimeout` example there and add a comment above the `setTimeout` that says something like: 

```The `dismiss` method can also be used to dismiss `ion-loading` after a long running task such as a network request.```

Comment (omar2205) [reply core/src/components/loading/usage/react.md:18]: we could add a comment to mention it
`setTimeout(dismiss, 2000) // simulate a dynamic dismissal of the IonLoading`

**Review by liamdebeasi — APPROVED** [review]
Great job!

**Comment by liamdebeasi** [comment]
Merged. Thank you!

**Pull request merged**